admin/includes/functions/general.php

```diff
@@ -3683,12 +3683,16 @@ function zen_get_ip_address() {
   /**
    * sanitize for validity as an IPv4 or IPv6 address
    */
-  $ip = preg_replace('~[^a-fA-F0-9.:%/]~', '', $ip);
+    $original_ip = $ip;
+    $ip = filter_var((string)$ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV6 | FILTER_FLAG_IPV4);
 
   /**
-   *  if it's still blank, set to a single dot
+   *  If it's an invalid IP, set the value to a single dot and issue a notification.
    */
-  if (trim($ip) == '') $ip = '.';
+  if ($ip === false) {
+      $ip = '.';
+      $GLOBALS['zco_notifier']->notify('NOTIFY_ZEN_ADMIN_INVALID_IP_DETECTED', $original_ip);
```

Comment:
Why isn't the notifier being allowed to modify the `$ip` variable by sending it as an editable value?

Reply:
Because the function's doing the work; no need to obfuscate debugging with some unknown alterations.

Reply:
By offering the value to be modified then the function's response can be honed to support the environment *without* touching the code which would seem desirable.

Reply:
Not to me; I'd consider a call to this function as the _single-source-of-truth_.  Noting that, upon receipt of the notification, you can do whatever is needed to "hone" the environment.

Reply:
However, this change has also revised the operation of the function.  Formerly, it ensured that the provided response met the criteria of characters acceptable in an ip address without regard of the type of ip address that was being detected.  

The use of `filter_var` has changed the response to a type of ip address that matches a set of hard-coded criteria or a dot if not acceptable.  That criteria has become IPV4 or IPV6 without the ability to provide back from the function the value that was actually collected, to refine what is considered acceptable or to take action on the value determined even if it met the criteria.  The response no longer provides the detected ip address (made of ip address permissible values) but instead only the ip address that has been allowed through the original filter. Any other desired filter requires code modification affecting upgrade instead of just adding to the system outside of the core code.

Part of what led me to think/consider this was discussion about providing filterable information about "visitors" to the site.  Specifically the discussion was related to spiders; however, this area could provide an additional operator on which to act.

Reply:
> However, this change has also revised the operation of the function. Formerly, it ensured that the provided response met the criteria of characters acceptable in an ip address without regard of the type of ip address that was being detected.

Erm, the comment prior to the current clause reads:
```
    /**
     * sanitize for validity as an IPv4 or IPv6 address
     */
```
I take that, as do the built-in users of the function, as ensuring that the IP address returned is going to be either a valid IPv4/IPb6 address or an (uncommented) `.` otherwise.  An interested observer has the opportunity to grab the invalid value to do something with it if needed.